pkg/deploy/generator/resources.go

```diff
@@ -1089,6 +1078,116 @@ EOF
 
 chcon -R system_u:object_r:var_log_t:s0 /var/opt/microsoft/linuxmonagent
 
+cat >/usr/lib/systemd/system/download-mdm-credentials.service <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Service]
+Type=oneshot
+ExecStart=/usr/local/bin/download-mdm-credentials.sh
+EOF
+
+cat >/usr/lib/systemd/system/download-mdm-credentials.timer <<'EOF'
+[Unit]
+Description=Periodic mdm credentials refresh
+
+[Timer]
+OnBootSec=0min
+OnCalendar=0/12:00:00
+AccuracySec=5s
+
+[Install]
+WantedBy=timers.target
+EOF
+
+cat >/usr/local/bin/download-mdm-credentials.sh <<EOF
+#!/bin/bash
+set -eu
+
+NEW_CERT_FILE="/etc/temp-mdm.pem"
+CURRENT_CERT_FILE="/etc/mdm.pem"
+
+az login -i
+az account set -s "$SUBSCRIPTIONID"
+
+rm -f \$NEW_CERT_FILE
+
+for attempt in {1..5}; do
+  az keyvault certificate download --file \$NEW_CERT_FILE --id "https://$KEYVAULTPREFIX-svc.vault.azure.net/secrets/rp-mdm" && break
+  if [[ \$attempt -lt 5 ]]; then sleep 10; else exit 1; fi
+done
+az logout
```

Comment:
what is the benefit of doing `az logout` in the first place? If the machine is comprimised they can just run az login and have access to the identity. 
We can solve the race by starting them on different schedules with 6 hours between the scripts

Reply:
> We can solve the race by starting them on different schedules with 6 hours between the scripts

Don't do hacks like that please.

az logout is a nice to have, rather than essential.

What is necessary is to isolate the .azure directory used by the scripts such that if anyone SSHes to the RP for any reason and messes with the az configuration, it does not affect the scripts.

Reply:
@jim-minter 

> Don't do hacks like that please 

Why this is a hack?

First of all, all of this is already a big hack, it is just that we do not have much choice about this. 

Second, I don't see anything "hacky" about spreading out cron jobs in general. If we deploy RP once in a month or so, what difference does it make if mdsd downloads certs at 6am/6pm and mdm downloads certs at, say, 9am/9pm? As long as each of them meets in the requirement to have certs refreshed within 24 hours I don't see an importance of this to be pinned to some arbitrary time when RP has been started or to each other.
And at startup they run sequentially. This will solve the issue with az login/logout race.

Reply:
Discussed via bluejeans.  The environment variable to set is AZURE_CONFIG_DIR.